test/integration/src/test/java/io/pravega/test/integration/endtoendtest/EndToEndIndexSegmentTest.java

```diff
@@ -131,22 +131,20 @@ public void testTruncationOfIndexSegment() throws Exception {
         LocalController controller = (LocalController) controllerWrapper.getController();
 
         Map<Long, Long> streamCutPositions = new HashMap<>();
-        streamCutPositions.put(0L, 81L);
+        streamCutPositions.put(0L, 108L);
 
         StreamSegments streamSegments = controller.getCurrentSegments(scope, streamName).join();
         Collection<Segment> segments = streamSegments.getSegments();
-        segments.forEach(segment -> {
-            //Validating starting offset of the main and index segment before truncation
-            assertEquals(0, store.getStreamSegmentInfo(segment.getScopedName(), TIMEOUT).join().getStartOffset());
-            assertEquals(0, store.getStreamSegmentInfo(NameUtils.getIndexSegmentName(segment.getScopedName()), TIMEOUT).join().getStartOffset());
-        });
+        assertEquals(1, segments.size());
+        Segment segment = segments.iterator().next();
+        //Validating starting offset of the main and index segment before truncation
+        assertEquals(0, store.getStreamSegmentInfo(segment.getScopedName(), TIMEOUT).join().getStartOffset());
+        assertEquals(0, store.getStreamSegmentInfo(NameUtils.getIndexSegmentName(segment.getScopedName()), TIMEOUT).join().getStartOffset());
 
         controller.truncateStream(stream.getScope(), stream.getStreamName(), streamCutPositions).join();
-
-        segments.forEach(segment -> {
-            //Validating starting offset of the main and index segment after truncation
-            assertEquals(81, store.getStreamSegmentInfo(segment.getScopedName(), TIMEOUT).join().getStartOffset());
-            assertEquals(48, store.getStreamSegmentInfo(NameUtils.getIndexSegmentName(segment.getScopedName()), TIMEOUT).join().getStartOffset());
-        });
+        //Validating starting offset of the main and index segment after truncation
+        assertEquals(108L, store.getStreamSegmentInfo(segment.getScopedName(), TIMEOUT).join().getStartOffset());
+        long indexOffset = store.getStreamSegmentInfo(NameUtils.getIndexSegmentName(segment.getScopedName()), TIMEOUT).join().getStartOffset();
+        assertTrue("Index was: " + indexOffset, indexOffset == 24 || indexOffset == 48);
```

Comment:
Do you think this assertion may be flakey? Here we are ensuring that there will be max two events in index segment corresponding to four events in main segment.

Reply:
Updated